README
Use pari/gp to achieve parallel pollard_rho_algorithm


pari-2.5.4/CHANGES
# $Id$
Bug numbers refer to the BTS at http://pari.math.u-bordeaux.fr/Bugs/

[last column crossreferences current development release 2.6.0]
Done for version 2.5.4 (released 13/05/2013):

  Fixed
    1- sin'(1) with default(compatible,3) caused a crash                   [F97]
    2- rare corruption in ECM [#1240] factorint(,1) => SEGV                [F98]
    3- add missing call to normalize() in lift(t_POL/t_SER) [#1359]        [F99]
BA  4- listput was not SIGINT safe                                        [F100]
BA  5- znorder(Mod(5,16),200) -> 5 instead of 4                           [F101]
JD  6- galoisinit(x^3+x^2-2*x-1) -> uninitialized read [#1389]            [F106]
BA  7- polhermite(66) -> corrupted result [#1393]                         [F107]
    8- polred() could return non-squarefree polynomials [#1395]
BA  9- galoisfixedfield() could return a wrong result for polynomials     [F109]
       with large roots [#1406]
   10- bnrdisc could return a wrong result if the modulus had more than   [F110]
       2 prime divisors [#1399]
   11- y;z;(x^2+y)*w/z/x -> significant pointers lost [#1248]             [F113]
   12- thue(thueinit(x^3-1493,1),3) -> error                              [F115]
   13- typo in modr_safe -> precision error [#1413]                       [F116]

[last column crossreferences current development release 2.6.0]
Done for version 2.5.3 (released 03/10/2012):

  Fixed
    1- padicappr(x^2+1+O(3), -1+O(5^10)) --> no error                      [F34]
    2- sqrtn(1+O(5),5) -> error, ispower(1+O(5),5) -> error [#1261]        [F65]
PB  3- printf("%.6f", 5E-5) -> SEGV [#1328]                                [F83]
BA  4- diffop(Mod(y,x^2-y),[y],[1]) -> wrong result                        [F84]
BA  5- calling a GP function with a lot of omitted arguments could crashes [F85]
GH  6- missing solutions in thue() [#951]                                  [F86]
    7- t_RFRAC == t_RFRAC could return a wrong result [#1339]              [F87]
    8- obscure bug in polroots() (sage #13314)                             [F88]
BA  9- printf() did not flush output                                       [F90]
   10- polrootsff(2*x+1,2,y) -> SEGV [#1350]                               [F93]
BA 11- elltors() was using an insufficient precision  [#1326]
   12- resultant(x,x,y) -> 0 and related problems                          [F94]
   13- thue((x^4+1)^2, 4) -> no solution                                   [F95]
BA 14- idealramgroups() and idealfrobenius() did not accept a bnf          [F96]

[last column crossreferences current development release 2.6.0]
Done for version 2.5.2 (released 20/06/2012):

  Fixed
    1- [from 2.5.1] substvec slowdown                                      [F25]
    2- logfile: strip properly color escape sequences [#1225]              [C18]
BA  3- ispower(x < 0,, &n): n could have the wrong sign [#1302]            [F45]
    4- mathnf(..., 4) didn't remove 0 columns from HNF [#1271]             [F61]
JD  5- possible SEGV in pari_init_opts [#1264]                             [F62]
BA  6- ellorder(E,P*Mod(1,p)) did not work if E was defined over Z         [F63]
    7- parimacro.tex: pdftex --output-format dvi  would produce a PDF      [F66]
BA  8- pari_close failed to free all allocated memory [#1297]              [F67]
    9- issquarefree(0) => error [#1304]                                    [F71]
   10- sqrt(0.E-97-1.12-97*I) -> div by 0 [#1309]                          [F76]
PH 11- vecsort0 was not valid C code (miscompiled by gcc 4.7)
BA 12- FpX_gcd(0,x) -> div by 0                                            [F77]
   13- x; y; p; q; subst(Mod(q,p), q, x + y) --> Mod(1, p)*x [#1321]       [F78]

[last column crossreferences current development release 2.6.0]
Done for version 2.5.1 (released 06/02/2012):

  Fixed
    1- Euclidean chains were confused by t_POLs with t_INTMOD coeffs       [F01]
       [ because a 0 polynomial could have lg() == 3 ]
    2- remi2n didn't allow negative arguments [native kernel] [#1215]      [F04]
    3- qfbred(t_QFR) wrong when frac( sqrt(D) ) > 0.5                      [F06]
    4- problems with diagnostic messages when changing a 'default', e.g \o3[F08]
    5- divru() could call bfffo(0) => overflow in expo()                   [F09]
AMe 6- (t_FRAC<0) * t_INTMOD => wrong result                               [F10]
    7- trace(t_POL or t_SER) could yield unormalized object [0 lead. term] [F11]
LGr 8- off-by-1 error in primepi() when argument is = primelimit           [F12]
    9- crash when default(prettyprinter,...) points to non-existent cmd    [F15]
   10- qfperfection(): wrong result in unlucky cases [#1223]               [F16]
BA 11- allocatemem() did not reset parse error [#1226]                     [F17]
BA 12- (x->vector(1,j,x))(1) --> significant pointers lost [#1211]         [F18]
VL 13- [makefile] Incorrect -I and -L flags order [#1212]                  [F19]
   14- nfbasis / nfdisc(f,, partial factorization) would sometimes spend a [F20]
       lot of time trying to factorize a discriminant [#1227]
BA 15- missing consistency checks in conjvec [#1231]                       [F22]
   16- wrong result in bezoutres [#1233]                                   [F24]
   17- wrong sign in polresultant() [#1234]                                [F25]
   18- subst(Y/X,X,x) -> 0                                                 [F27]
PB 19- polrootsff(x^2-x-ffgen((v^2+1) * Mod(1,3))) -> SEGV [#1241]         [F28]
BA 20- functions ending by the x[y]=z construct could corrupt the stack.   [F30]
BA 21- test-kernel and test-ploth did not work                             [F35]
BA 22- for(i=1,10^7,) + SIGINT -> SEGV [#1249]                             [F37]
BA 23- mateigen precision error was not trappable.                         [F40]
   24- broken rnfeltup [#1256]                                             [F42]
   25- x===y was always wrong for x,y t_SER                                [F43]

f2Mul.s
	.set noreorder
	.set volatile
	.set noat
.text
	.align 3
	.globl f2Mul
	.ent f2Mul
f2Mul:
        lda     $28,0x1111($31)
        srl	$17,1,$1 
        ldah    $28,0x1111($28)
        srl	$17,2,$4 
        sll     $28,32,$2
        lda	$30,-48($30)
        bis     $2,$28,$2
        srl	$18,2,$7 
        srl	$18,1,$21 
        bis	$31,$31,$31 
        and	$17,$2,$3 
        srl	$17,3,$17 
        and	$18,$2,$8 
        srl	$18,3,$18 
        and	$2,$4,$4 
        and	$2,$17,$17 
        and	$2,$7,$7 
        and	$2,$18,$18 
        stq	$26,0($30)
        and	$2,$21,$21 
        stq	$9,8($30)
        and	$2,$1,$1 
        stq	$10,16($30)
        xor	$8,$7,$20 
        stq	$11,24($30)
        xor	$4,$17,$9 
        xor	$7,$18,$10 
        xor	$21,$18,$22 
        mulq	$9,$10,$11 
        xor	$3,$4,$5 
        xor	$1,$17,$6 
        xor	$20,$22,$23 
        xor	$5,$6,$19 
        xor	$3,$1,$0 
        xor	$8,$21,$26 
        stq	$16,32($30)
        mulq	$19,$23,$24 
        stq	$11,40($30)
        and	$19,$23,$11 
        cmpeq	$11,$2,$11 
        umulh	$19,$23,$19 
        and	$5,$20,$23 
        cmpeq	$23,$2,$23 
        and	$24,$2,$24 
        mulq	$5,$20,$25 
        subq	$19,$11,$11 
        and	$11,$2,$11 
        ldq_u	$31,0($30)
        umulh	$5,$20,$5 
        and	$6,$22,$20 
        cmpeq	$20,$2,$20 
        and	$25,$2,$25 
        xor	$24,$25,$24 
        mulq	$6,$22,$27 
        subq	$5,$23,$5 
        and	$0,$26,$23 
        and	$5,$2,$5 
        cmpeq	$23,$2,$23 
        xor	$11,$5,$11 
        ldq_u	$31,0($30)
        umulh	$6,$22,$6 
        and	$3,$8,$22 
        cmpeq	$22,$2,$22 
        and	$27,$2,$27 
        xor	$24,$27,$24 
        s4addq	$27,$31,$27 
        addq	$24,$24,$24 
        xor	$25,$24,$24 
        xor	$24,$27,$24 
        mulq	$0,$26,$16 
        subq	$6,$20,$6 
        and	$6,$2,$6 
        xor	$11,$6,$11 
        s4addq	$6,$31,$6 
        addq	$11,$11,$11 
        ldq_u	$31,0($30)
        umulh	$0,$26,$0 
        xor	$5,$11,$5 
        and	$1,$21,$11 
        and	$9,$10,$26 
        cmpeq	$11,$2,$11 
        cmpeq	$26,$2,$26 
        xor	$5,$6,$5 
        and	$16,$2,$16 
        mulq	$3,$8,$19 
        bis	$31,$31,$31 
        subq	$0,$23,$0 
        and	$4,$7,$23 
        and	$0,$2,$0 
        cmpeq	$23,$2,$23 
        umulh	$3,$8,$3 
        and	$19,$2,$19 
        xor	$16,$19,$16 
        mulq	$1,$21,$20 
        subq	$3,$22,$3 
        and	$17,$18,$22 
        and	$3,$2,$3 
        cmpeq	$22,$2,$22 
        xor	$0,$3,$0 
        umulh	$1,$21,$1 
        and	$20,$2,$20 
        bis	$31,$31,$31 
        xor	$16,$20,$16 
        s4addq	$20,$31,$20 
        addq	$16,$16,$16 
        xor	$19,$20,$19 
        xor	$19,$16,$16 
        xor	$24,$16,$24 
        mulq	$4,$7,$8 
        subq	$1,$11,$1 
        and	$1,$2,$1 
        ldq_u	$31,0($30)
        s4addq	$1,$31,$6 
        xor	$0,$1,$0 
        ldq	$1,40($30)
        xor	$3,$6,$3 
        umulh	$9,$10,$9 
        addq	$0,$0,$0 
        and	$1,$2,$1 
        xor	$3,$0,$0 
        xor	$5,$0,$5 
        ldq	$3,32($30)
        and	$8,$2,$8 
        xor	$1,$8,$1 
        umulh	$4,$7,$4 
        bis	$31,$31,$31 
        subq	$9,$26,$9 
        ldq	$26,0($30)
        and	$9,$2,$9 
        umulh	$17,$18,$21 
        subq	$4,$23,$4 
        and	$4,$2,$4 
        xor	$9,$4,$9 
        mulq	$17,$18,$17 
        subq	$21,$22,$21 
        and	$21,$2,$21 
        xor	$9,$21,$9 
        s4addq	$21,$31,$21 
        addq	$9,$9,$9 
        xor	$4,$21,$4 
        xor	$4,$9,$4 
        ldq	$9,8($30)
        s4addq	$4,$31,$11 
        xor	$5,$4,$4 
        xor	$4,$11,$4 
        ldq	$11,24($30)
        s4addq	$4,$31,$4 
        ldq_u	$31,0($30)
        and	$17,$2,$2 
        xor	$0,$4,$0 
        xor	$1,$2,$1 
        s4addq	$2,$31,$2 
        addq	$1,$1,$1 
        xor	$8,$2,$2 
        xor	$2,$1,$1 
        ldq_u	$31,0($30)
        xor	$24,$1,$24 
        s4addq	$1,$31,$10 
        srl	$24,62,$6 
        xor	$24,$10,$10 
        srl	$1,60,$1 
        xor	$0,$6,$0 
        xor	$0,$1,$0 
        s4addq	$10,$31,$10 
        stq	$0,0($3)
        xor	$16,$10,$0 
        ldq	$10,16($30)
        lda	$30,48($30)
        ret	$31,($26),1
	.end f2Mul
